packages/cli/generation/ir-generator/src/filterExamples.ts

```diff
@@ -0,0 +1,336 @@
+import {
+    ExampleContainer,
+    ExampleEndpointCall,
+    ExampleInlinedRequestBodyProperty,
+    ExampleKeyValuePair,
+    ExampleNamedType,
+    ExampleObjectProperty,
+    ExamplePathParameter,
+    ExampleRequestBody,
+    ExampleResponse,
+    ExampleSingleUnionTypeProperties,
+    ExampleType,
+    ExampleTypeReference,
+    ExampleTypeReferenceShape,
+    ExampleTypeShape,
+    NameAndWireValue
+} from "@fern-api/ir-sdk";
+import { FilteredIr } from "./filtered-ir/FilteredIr";
+
+function filterExampleSingleUnionTypeProperties(
+    filteredIr: FilteredIr,
+    singleUnionTypeProperties: ExampleSingleUnionTypeProperties
+): ExampleSingleUnionTypeProperties | undefined {
+    return singleUnionTypeProperties._visit<ExampleSingleUnionTypeProperties | undefined>({
+        samePropertiesAsObject: (s) => {
+            const filteredObject = filteredIr.hasTypeId(s.typeId)
+                ? {
+                      ...s,
+                      object: {
+                          properties: s.object.properties
+                              .filter((p) => filteredIr.hasProperty(p.originalTypeDeclaration.typeId, p.name.wireValue))
+                              .map((p) => ({
+                                  ...p,
+                                  value: filterExampleTypeReference(filteredIr, p.value)
+                              }))
+                              .filter((p) => p.value !== undefined) as ExampleObjectProperty[]
```

Comment:
Why do we have to do a cast here: `as ExampleObjectProperty[]` ?

Reply:
It still thinks there are undefined values without the cast, this is a pretty common thing

Reply:
whenever i've encountered this pattern (having to filter the nulls at the end) I've typically found that its better to use a normal `for` loop

Reply:
Yeah! I think it's a matter of preference 
I like the map and filter but I can change it if you'd prefer

Reply:
Yeah we've just stayed away from any usage of `as` to prevent any weird bugs and have the compiler help as much as possible. (the situation I'm concerned about is what if someone modifies the map but the compiler doesn't give them an error because of the cast)

Reply:
Cool, removed them